File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        a,n=[],len(tokens)
        for i in range(n):
            if tokens[i]=='+' or tokens[i]=='-' or tokens[i]=='*' or tokens[i]=='/':
                x=int(a.pop())
                y=int(a.pop())
                if tokens[i]=='+':
                    a.append(y+x)
                elif tokens[i]=='-':
                    a.append(y-x)
                elif tokens[i]=='*':
                    a.append(y*x)
                else:
                    a.append(y/x)
            else:
                a.append(tokens[i])
        return int(a[0])

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> s;
        auto isSign = [](string token) {
            return (token == "+" || token == "-" || token == "*" || token == "/");
        };
        for (auto token : tokens) {
            if (isSign(token)) {
                int num2 = s.top();
                s.pop();
                int num1 = s.top();
                s.pop();
                if (token == "+") {
                    s.push(num1 + num2);
                } else if (token == "-") {
                    s.push(num1 - num2);
                } else if (token == "*") {
                    s.push(num1 * num2);
                } else if (token == "/") {
                    s.push(num1 / num2);
                }
            } else {
                s.push(stoi(token));
            }
        }
        return s.top();
    }
};